### DISCUSSION TITLE
How to use Tooltip Tooltip Dialog and Button together?

### DISCUSSION THREAD

**Discussion by pavelbe** [shadcn-ui/ui#2170] [Q&A]
**Cannot combine a dialog box with a button and a tooltip.**

There is an error in the browser console:

app-index.js:32 Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
Check the render method of `SlotClone`.
	
And I can't fix it.
Help please!!!

Here is my code:
`
// src/components/dashboard/feedback/BugReportButton.tsx
"use client"
import { useState } from "react"
import * as React from "react"
import { DashboardIcons } from "@/icons/dashboardIcons"
import { Button } from "@/ui/button"
import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from "@/ui/tooltip"
import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogHeader,
  DialogTitle,
  DialogTrigger,
} from "@/components/ui/dialog"

export function BugReportButton() {
  const [isOpen, setIsOpen] = useState(false)

  const bugBtn = (
    <TooltipProvider>
      <Tooltip>
        <TooltipTrigger asChild>
          <Button
            variant='ghost'
            size='icon'
            className='focus-button relative size-9 rounded-full shadow-[0_0_0_2px_#dededf] transition-all duration-300 ease-in hover:bg-[#eaedf1] dark:shadow-[0_0_0_2px_#2c3139] dark:hover:bg-[#2c3139]'
            aria-label='Send an error message'
            onClick={() => setIsOpen(true)} // This will now only handle opening the dialog
          >
            <DashboardIcons.bug />
            <span className='sr-only'>Send an error message</span>
          </Button>
        </TooltipTrigger>
        <TooltipContent side='right'>Send an error message</TooltipContent>
      </Tooltip>
    </TooltipProvider>
  )

  return (
    <Dialog open={isOpen} onOpenChange={setIsOpen}>
      <DialogTrigger asChild>{bugBtn}</DialogTrigger>
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Please leave your application</DialogTitle>
          <DialogDescription>Our experts will contact you shortly. Thank you!</DialogDescription>
        </DialogHeader>
      </DialogContent>
    </Dialog>
  )
}
`

**Comment by Varun-Chakraborty** [comment]
if your issue is just inability to reference functional component then try using React.forwardRef()
